### ISSUE TITLE
`unnecessary_doc_comment`, `inaccessible_doc_comment`, or something similar

### ISSUE THREAD

**Issue by smoelius** [trailofbits/dylint#1575]
The Rust compiler has an [`unused_doc_comments`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.UNUSED_DOC_COMMENTS.html) lint that warns about doc comments applied to [nodes of the wrong kind](https://github.com/rust-lang/rust/blob/5e17a2a91dd7dbefd8b4a1087c2e42257457deeb/compiler/rustc_lint/src/builtin.rs#L902). But it does not warn about nodes of the right kind that are simply inaccessible.

Consider the program below ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=5a8ebaceacfd0ca9c934bb7e09f00e00)). `unused_doc_comments` warns about doc comment preceding the statement, but not the one preceding the constant.
```rs
fn main() {
    /// A constant
    const GREETING: &str = "Hello, world!";
    /// A statement
    println!("{GREETING}");
}
```
